<a href="https://colab.research.google.com/github/Kainat209546/flyrank-ml-internship/blob/main/Copy_of_w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Kainat209546/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [6]:
import os
import sys
import subprocess

REPO_URL = "https://github.com/Kainat209546/flyrank-ml-internship"
REPO_DIR = "flyrank-ml-internship"

if not os.path.isdir(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)

os.chdir(REPO_DIR)

print("Current directory:", os.getcwd())

Current directory: /flyrank-ml-internship/flyrank-ml-internship


## Research Question

**Question**

Which content pages should be reviewed first for refresh based on observable search and content signals?

### Decision

Determine which pages deserve priority for review.

### Unit of Analysis

One content page.

### Output

A ranked list of pages with refresh priority scores and reason codes.

### Action

A content manager can review, update, expand, monitor, or improve the highest-ranked pages first.

### Cost of a Wrong Recommendation

If a page is incorrectly ranked as high priority, valuable review time may be wasted.

If an important declining page is missed, traffic and visibility may continue to decrease.

### Why Machine Learning Helps

Many search and engagement signals interact together. Machine learning can combine these signals better than a simple hand-written rule and produce a more useful ranking for reviewers.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Question**

Which content pages should be reviewed first for refresh based on observable search and content signals?

### Decision

Determine which pages deserve priority for review.

### Unit of Analysis

One content page.

### Output

A ranked list of pages with refresh priority scores and reason codes.

### Action

A content manager can review, update, expand, monitor, or improve the highest-ranked pages first.

### Cost of a Wrong Recommendation

If a page is incorrectly ranked as high priority, valuable review time may be wasted.

If an important declining page is missed, traffic and visibility may continue to decrease.

### Why Machine Learning Helps

Many search and engagement signals interact together. Machine learning can combine these signals better than a simple hand-written rule and produce a more useful ranking for reviewers.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [7]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total Pages:", len(df))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Average Impressions (90d):", round(df["impressions_90d"].mean(), 2))

print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())

Total Pages: 30000
Average CTR: 0.511
Average Impressions (90d): 5200.37

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


## Quick Look at the Data

I explored the starter dataset to better understand the problem before defining my research question.

Key observations:

- The dataset contains **30,000 content pages**.
- The average **CTR (Click-Through Rate)** is **0.511**.
- The average **90-day impressions** is **5,200.37**.
- The largest group of pages is **declining ("down")**, with **16,262 pages**, followed by **stable (5,962)** and **up (4,388)**.

These observations suggest that many pages are already experiencing declining performance. Therefore, building a system to prioritize which pages should be reviewed first could help content teams make better decisions.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project identifies patterns that are associated with content refresh opportunities.

The results do not prove that refreshing a page will increase traffic.

The recommendations are intended to support human decision making rather than replace it.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.